libs/partners/ai21/tests/integration_tests/test_semantic_text_splitter.py

```diff
@@ -0,0 +1,143 @@
+    "Pokémon world; discover and catch each Pokémon species in order to complete their"
+    "Pokédex; and train a team of up to six Pokémon at a time and have them engage "
+    "in battles.\nMost Pokémon can be caught with spherical devices known as Poké "
+    "Balls.\nOnce the opposing Pokémon is sufficiently weakened, the Trainer throws "
+    "the Poké Ball against the Pokémon, which is then transformed into a form of "
+    "energy and transported into the device.\nOnce the catch is successful, "
+    "the Pokémon is tamed and is under the Trainer's command from then on.\n"
+    "If the Poké Ball is thrown again, the Pokémon re-materializes into its "
+    "original state.\nThe Trainer's Pokémon can engage in battles against opposing "
+    "Pokémon, including those in the wild or owned by other Trainers.\nBecause the "
+    "franchise is aimed at children, these battles are never presented as overtly "
+    "violent and contain no blood or gore.[I]\nPokémon never die in battle, instead "
+    "fainting upon being defeated.[20][21][22]\nAfter a Pokémon wins a battle, it "
+    "gains experience and becomes stronger.[23] After gaining a certain amount of "
+    "experience points, its level increases, as well as one or more of its "
+    "statistics.\nAs its level increases, the Pokémon can learn new offensive "
+    "and defensive moves to use in battle.[24][25] Furthermore, many species can "
+    "undergo a form of spontaneous metamorphosis called Pokémon evolution, and "
+    "transform into stronger forms.[26] Most Pokémon will evolve at a certain level, "
+    "while others evolve through different means, such as exposure to a certain "
+    "item.[27]\n"
+)
+
+
+def test_invoke__split_text_to_document() -> None:
+    segmentation = AI21SemanticTextSplitter()
+    segments = segmentation.split_text_to_documents(source=TEXT)
+    assert len(segments) > 0
+    for segment in segments:
+        assert segment.page_content is not None
+        assert segment.metadata is not None
+
+
+def test_invoke__split_text() -> None:
+    segmentation = AI21SemanticTextSplitter()
+    segments = segmentation.split_text(source=TEXT)
+    assert len(segments) > 0
+
+
+def test_invoke__split_text__chunk_size() -> None:
+    segmentation_no_merge = AI21SemanticTextSplitter()
+    segments_no_merge = segmentation_no_merge.split_text(source=TEXT)
+    segmentation_merge = AI21SemanticTextSplitter(chunk_size=1000)
+    segments_merge = segmentation_merge.split_text(source=TEXT)
+    # Assert that a merge did happen
+    assert len(segments_no_merge) > len(segments_merge)
+    reconstructed_text_merged = "".join(segments_merge)
+    reconstructed_text_non_merged = "".join(segments_no_merge)
+    # Assert that the merge did not change the content
+    assert reconstructed_text_merged == reconstructed_text_non_merged
+
+
+def test_invoke__split_text__chunk_size__small_chunk_size() -> None:
```

Comment:
I would rename this to `test_splt_text__when_chunk_size_is_too_small__should_return_non_merged_segments` or something like this that would shed a little more light as to what makes this use case different